src/darwin/Framework/CHIP/ServerEndpoint/MTRServerAttribute.h

```diff
@@ -0,0 +1,65 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRBaseClusters.h>
+#import <Matter/MTRDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A representation of an attribute implemented on a server cluster by an
+ * MTRDeviceController.  An attribute has an identifier and a value, and may or
+ * may not be writable.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRServerAttribute : NSObject <NSCopying>
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Initialize as a readonly attribute.  The value is a data-value as documented
+ * in MTRBaseDevice.h.
+ *
+ * Will fail if the attribute ID is not valid per the Matter specification or
+ * the attribute value is not a valid data-value.
+ *
+ * requiredPrivilege is the privilege required to read the attribute.
+ */
+- (nullable instancetype)initReadonlyAttributeWithID:(NSNumber *)attributeID initialValue:(NSDictionary<NSString *, id> *)value requiredPrivilege:(MTRAccessControlEntryPrivilege)requiredPrivilege;
+
+/**
+ * Change the value of the attribute to a new value.  The value is a data-value as documented
+ * in MTRBaseDevice.h.
+ *
+ * Will fail if the attribute is not a valid data-value or if the attribute is
+ * on an endpoint that used to be defined on a controller and isn't any longer,
+ * whether because it was removed or because the controller was shut down.
+ */
+- (BOOL)setValue:(NSDictionary<NSString *, id> *)value;
```

Comment:
Setting a value should work OK even if the attribute is no longer "attached" ? It would just update the property of the object itself. Seems like it should be possible to re-attach an endpoint to a new controller as well (as long as it's only "live" on one controller at a time)

Reply:
> Setting a value should work OK even if the attribute is no longer "attached" ?

We _could_ allow that, but it likely indicates programmer error in that the set would get silently ignored in practice.  Note that an attribute can get detached if the controller is shut down, which can be done by a different party from the part that is manipulating the attribute.

> Seems like it should be possible to re-attach an endpoint to a new controller as well

It's actually quite complicated to do that and I would rather not try to allow it.

Reply:
Looked at this some more and did this.